File: JAVA/PokerProtocol/src/protocol/commands/lobby/response/JoinTableResponse.java
package protocol.commands.lobby.response;

import java.util.StringTokenizer;

import protocol.commands.Command;
import protocol.commands.ICommand;

public class JoinTableResponse implements ICommand
{
    private final int m_tableID;
    private final String m_playerName;
    private final int m_noSeat;
    public static String COMMAND_NAME = "lobbyJOIN_TABLE_RESPONSE";
    
    public JoinTableResponse(StringTokenizer argsToken)
    {
        m_playerName = argsToken.nextToken();
        m_tableID = Integer.parseInt(argsToken.nextToken());
        m_noSeat = Integer.parseInt(argsToken.nextToken());
    }
    
    public JoinTableResponse(String pName, int tName, int seat)
    {
        m_playerName = pName;
        m_tableID = tName;
        m_noSeat = seat;
    }
    
    @Override
    public String encodeCommand()
    {
        final StringBuilder sb = new StringBuilder();
        sb.append(JoinTableResponse.COMMAND_NAME);
        sb.append(Command.L_DELIMITER);
        sb.append(m_playerName);
        sb.append(Command.L_DELIMITER);
        sb.append(m_tableID);
        sb.append(Command.L_DELIMITER);
        sb.append(m_noSeat);
        sb.append(Command.L_DELIMITER);
        return sb.toString();
    }
    
    public String getPlayerName()
    {
        return m_playerName;
    }
    
    public int getTableID()
    {
        return m_tableID;
    }
    
    public int getNoSeat()
    {
        return m_noSeat;
    }
}

File: C#/PokerProtocol/Commands/Lobby/Response/JoinTableResponse.cs
using System;
using System.Collections.Generic;
using System.Text;
using EricUtility;
using EricUtility.Networking.Commands;

namespace PokerProtocol.Commands.Lobby.Response
{
    public class JoinTableResponse : AbstractLobbyCommandResponse<JoinTableCommand>
    {

        protected override string CommandName
        {
            get { return COMMAND_NAME; }
        }
        public static string COMMAND_NAME = "lobbyJOIN_TABLE_RESPONSE";
        private readonly int m_NoSeat;
        public int NoSeat
        {
            get { return m_NoSeat; }
        } 


        public JoinTableResponse(StringTokenizer argsToken)
            : base(new JoinTableCommand(argsToken))
        {
            m_NoSeat = int.Parse(argsToken.NextToken());
        }

        public JoinTableResponse(JoinTableCommand command, int seat)
            : base(command)
        {
            m_NoSeat = seat;
        }

        public override void Encode(StringBuilder sb)
        {
            base.Encode(sb);
            Append(sb, m_NoSeat);
        }
    }
}
